### PR TITLE
Bug fixes

### LINKED ISSUE #288: Client needs more obvious 'file did not upload' exit status
While attempting to upload a file, if failure occurs, it's not quite clear that things did not complete.

Please add something a bit more obvious as the last line of the upload process that clearly indicates  this. Example "[error] Failed to upload file '$file'."

Potentially another bug report, but shouldn't a new attempt be tried after X failures to add shard metadata?

```
 [Mon Jul 25 2016 11:09:19 GMT-0500 (CDT)] [info]   Generating encryption key...
 [Mon Jul 25 2016 11:09:19 GMT-0500 (CDT)] [info]   Encrypting file "/data/fileData.data"
 [Mon Jul 25 2016 11:09:19 GMT-0500 (CDT)] [info]   Encryption complete!
 [Mon Jul 25 2016 11:09:19 GMT-0500 (CDT)] [info]   Creating storage token...
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Storing file, hang tight!
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Trying to upload shard /tmp/2c3dc37ab167 index 0
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Hash for this shard is: 57fa06124ce0d13d5b5e1db9d0be800103d508ce
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Audit generation for shard done.
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Waiting on a storage offer from the network...
 [Mon Jul 25 2016 11:09:20 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:09:36 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:09:51 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:10:07 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:10:22 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:10:38 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:10:53 GMT-0500 (CDT)] [info]   Adding shard metadata for 57fa06124ce0d13d5b5e1db9d0be800103d508ce to frame
 [Mon Jul 25 2016 11:11:08 GMT-0500 (CDT)] [warn]   Failed to upload shard...
 [Mon Jul 25 2016 11:11:08 GMT-0500 (CDT)] [error]  No storage offers were received
 [Mon Jul 25 2016 11:11:08 GMT-0500 (CDT)] [info]   Cleaning up...
 [Mon Jul 25 2016 11:11:08 GMT-0500 (CDT)] [info]   Finished cleaning!
```

### LINKED ISSUE #299: Uploading an empty file results in stack trace
### Package Versions

Replace the values below using the output from `npm list storj`. Use ```npm list -g storj``` if installed globally.

```
├── storj@2.0.7
└─┬ storj-bridge-client@0.4.4
  └── storj@0.6.16
```

Replace the values below using the output from `node --version`.

```
v5.10.1
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

```
I would expect core to let me know that I'm trying to upload a 0 byte file instead of returning a stack trace.
```

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

```
When uploading a 0 byte file I get a stack trace...

storj@monitoring-01:~/scripts$ storj upload-file 57a0cb7bee1e331f035a11a5 /tmp/test_download_file.10m
 [...]  > Enter your passphrase to unlock your keyring  >  ********

 [Tue Aug 02 2016 12:35:56 GMT-0400 (EDT)] [info]   Generating encryption key...
 [Tue Aug 02 2016 12:35:56 GMT-0400 (EDT)] [info]   Encrypting file "/tmp/test_download_file.10m"
 [Tue Aug 02 2016 12:35:56 GMT-0400 (EDT)] [info]   Encryption complete!
 [Tue Aug 02 2016 12:35:56 GMT-0400 (EDT)] [info]   Creating storage token...
 [Tue Aug 02 2016 12:35:57 GMT-0400 (EDT)] [info]   Storing file, hang tight!
/opt/storj/.nvm/versions/node/v5.10.1/lib/node_modules/storj/lib/filedemuxer.js:105
  this._currentOutput.push(bytes);
                     ^

TypeError: Cannot read property 'push' of undefined
    at FileDemuxer._pushBytesToOutput (/opt/storj/.nvm/versions/node/v5.10.1/lib/node_modules/storj/lib/filedemuxer.js:105:22)
    at FileDemuxer._closeCurrentOutput (/opt/storj/.nvm/versions/node/v5.10.1/lib/node_modules/storj/lib/filedemuxer.js:113:8)
    at FileDemuxer._handleSourceEnded (/opt/storj/.nvm/versions/node/v5.10.1/lib/node_modules/storj/lib/filedemuxer.js:92:8)
    at emitNone (events.js:85:20)
    at ReadStream.emit (events.js:179:7)
    at endReadableNT (_stream_readable.js:913:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

### Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

1. Try to upload a 0 byte file

### PR DESCRIPTION
**Pull request by aleitner**
Closes https://github.com/Storj/storjshare-cli/issues/140
Closes #299 
Closes #288

### CODE REVIEW

File: lib/filedemuxer.js [hunk 1]
```diff
@@ -59,6 +59,11 @@ inherits(FileDemuxer, EventEmitter);
  * @private
  */
 FileDemuxer.prototype._openStream = function() {
+  if (this._fileSize === 0) {
+    this._currentOutput = new stream.Readable({ read: utils.noop });
+    this.emit('error', new Error('0 byte file size is for idiots!'));
```
Comment (bookchin) [inline lib/filedemuxer.js]: @aleitner i agree, but maybe we shouldn't insult the user :)

File: test/filedemuxer.unit.js [hunk 2]
```diff
@@ -93,6 +94,19 @@ describe('FileDemuxer', function() {
...
 
+    it('should emit an error if file size is 0B', function(done) {
+      this.timeout(6000);
+
+      fs.closeSync(fs.openSync(filePathEmpty, 'w'));
+      
+      var dmx = new FileDemuxer(filePathEmpty);
+
+      dmx.on('error', function(err) {
+        expect(err.message).to.equal('0 byte file size is for idiots!');
```
Comment (bookchin) [inline test/filedemuxer.unit.js]: You even wrote a test? :sob:

**Comment by NicoNet** [comment]
Shouldn't we be able to store files of 0 bytes?  All files should be stored, even 0B files.

**Comment by bookchin** [comment]
Could you explain a scenario where it would be desirable to do so?

The problem here is that we cannot encrypt, shard, or transfer a file that has no content. We don't work with file descriptors, just raw bytes. If there are no bytes, then there is no file in Storj land.

**Comment by NicoNet** [comment]
I can't think of any good, highly legitimate reasons to do so.  But, I do agree that the user shouldn't be insulted with the error message.

File: lib/bridgeclient.js [hunk 3]
```diff
@@ -823,6 +823,7 @@ BridgeClient.prototype._request = function(method, path, params, callback) {
     }
 
     if (res.statusCode >= 400) {
+      console.log(body.error);
```
Comment (bookchin) [inline lib/bridgeclient.js]: Remember to remove this `console.log`

File: lib/filedemuxer.js [hunk 4]
```diff
@@ -59,6 +59,11 @@ inherits(FileDemuxer, EventEmitter);
  * @private
  */
 FileDemuxer.prototype._openStream = function() {
+  if (this._fileSize === 0) {
+    this._currentOutput = new stream.Readable({ read: utils.noop });
+    this.emit('error', new Error('File size cannot be 0 Bytes.'));
```
Comment (bookchin) [inline lib/filedemuxer.js]: I'd add a `return` in front of `this.emit('error', ...)` so we don't fall through and set up event listeners on the dead stream.

**Comment by bookchin** [comment]
@aleitner just added 2 more comments, then this is good to go :)

**Pull request merged**